### ISSUE TITLE
Easy Option to add some variables to all views?

### ISSUE THREAD

**Issue by benyanke** [laracasts/PHP-Vars-To-Js-Transformer#114]
In addition to controller-method specific variables, is there an easy way to also specify some variables to use on all pages? This would be used, for example, to pass options from config files to JS frontend apps.

**Comment by Luddinus** [comment]
You could load the javascript variables in a ViewComposer code.

**Comment by AlexCatch** [comment]
Yeah like @Luddinus said, create a `ViewServiceProvider` & in the boot method you can write something like this.

```php
View::composer('*', function ($view) {
    // JS facade used here
});
```

**Comment by generalProf** [comment]
**TLDR**
Make sure you assign the vars prepend to a rendering view that does not extend a main view, rather put a empty view at the bottom of your content area for the composer to be able to pass the data through.
(I could be wrong but i think this is a @extend blade problem by not having the content within the view be rendered but rather passed through a @section to the main view)

For anyone that is having a hard time getting the data through to the front end using a composer file. this is what helped me.

What I was doing wrong:
In my JavaScript config I set the attach view to a view that extends a main view, with a content section within it that get's passed to the main view.

What I did to fix it:
In the JavaScript I set a new view which gets rendered within the front end since i place this view within the main view (where html body tag structure is) and that worked.
